proposals/drafts/elections.rst

```diff
@@ -43,46 +255,502 @@ extras
+    Specifies the full text of the ballot measure as it appears on the ballot (string).
+
+info_uri
+    **optional**
+    Specifies a URI that links to additional information about the ballot measure (string).
+
+passage_threshold
+    **optional**
+    Specifies the threshold of votes the ballot measure needs in order to pass. The default is a simple majority, i.e., "50% plus one vote". Other common thresholds are "three-fifths" and "two-thirds".
+
+pro_statement
+    **optional**
+    Specifies a statement in favor of the referendum. It does not necessarily appear on the ballot (string).
+
+summary_text
+    **optional**
+    Specifies a short summary of the ballot measure that is on the ballot, below the title, but above the text.
+
+type
+    **optional**
+    Enumerated among "ballot-measure", "referendum", "initiative" and "other".
+
+other_type
+    **optional**
+    Allows for cataloging a new type of ballot measure option, when type is specified as "other" (string).
+
 
 CandidateContest
 ----------------
 
+A subclass of ``ContestBase`` for repesenting a contest among candidates competing for election to a public office.
+
 id
-    ocd identifier
+    Open Civic Data-style id in the format ``ocd-candidatecontest/{{uuid}}``.
 
-election_id
-    reference to the ``Election`` this contest is part of
+filing_deadline
+    **optional**
+    Specifies the date and time when a candidate must have filed for the contest for the office (datetime).
+
+is_runoff
+    Indicates a contest to decide a prior contest that ended with no candidate receiving a majority of the votes (boolean).
+
+is_unexpired_term
+    Indicates that the former public office holder vacated the post before serving a full term (boolean).
+
+number_elected
+    **optional**
+    Number of candidates that are elected in the contest, i.e. 'N' of N-of-M (integer).
+
+post_ids
+    References to the ``Posts`` representing the public offices for which the candidates are competing. If multiple, the primary post should be listed first, e.g., the id for the President post should be listed before the id for Vice-President.
+
+party_id
+    **optional**
+    If the contest is among candidates of the same political party, e.g., a partisan primary election, reference to that ``Party``.
+
+votes_allowed
+    **optional**
+    Maximum number of votes/write-ins per voter in this contest.
+
+
+PartyContest
+------------
+
+A subclass of ``ContestBase`` which describes a contest in which the possible ballot selections are all political parties. These could include contests in which straight-party selections are allowed, or party-list contests (although these are more common outside of the United States).
+
+id
+    Open Civic Data-style id in the format ``ocd-partycontest/{{uuid}}``.
+
+
+RetentionContest
+----------------
+
+A subclass of ``BallotMeasureContest`` that represents a contest where a candidate is retained in a position, e.g. a judicial retention or recall election.
```

Comment:
VIP's [BallotMeasureContest](http://vip-specification.readthedocs.io/en/release/built_rst/xml/elements/ballot_measure_contest.html) element, which corresponds to the data type which `RetentionContest` is subclassing, has a `BallotMeasureThreshold` text field. If you think we need something like that, please propose it on the [other PR](https://github.com/opencivicdata/docs.opencivicdata.org/pull/64).